dspace-api/src/main/java/org/dspace/app/util/AuthorizeUtil.java

```diff
@@ -601,4 +607,41 @@ public static void authorizeManageGroup(Context context, Group group) throws SQL
 
         throw new AuthorizeException("not authorized to manage this group");
     }
+
+    /**
+     * This method will return a boolean indicating whether the current user is allowed to register a new
+     * account or not
+     * @param context   The relevant DSpace context
+     * @param request   The current request
+     * @return          A boolean indicating whether the current user can register a new account or not
+     * @throws SQLException If something goes wrong
+     */
+    public static boolean authorizeNewAccountRegistration(Context context, HttpServletRequest request)
+        throws SQLException {
+        if (DSpaceServicesFactory.getInstance().getConfigurationService()
+                                 .getBooleanProperty("user.registration", true)) {
+            return AuthenticateServiceFactory.getInstance().getAuthenticationService()
+                                             .allowSetPassword(context, request, null);
```

Comment:
This won't work correctly because e.g. the LDAP `canSelfRegister` will return true due to auto-register, while that doesn't imply a new user can register explicitly. The `allowSetPassword` is currently the only method returning true only where expected. A new `canSelfRegisterManually` might be required instead

Reply:
@benbosman : I'm OK with us explaining this in more detail in a code comment. I was simply noting here that it's odd for a method named "authorizeNewAccountRegistration" to be checking whether you have permissions to **set your password**.  That jumped out as very odd to me, but it sounds like there's reasons for it.  Could we just describe those reasons in either the JavaDocs or in an inline comment here?

Reply:
This has been explained in code comments now